packages/engine/src/behave-graph/nodes/Profiles/Engine/Values/EntityNodes.ts

```diff
@@ -250,6 +268,75 @@ export const setEntityTransform = makeFlowNodeDefinition({
   }
 })
 
+export const useEntityTransform = makeEventNodeDefinition({
+  typeName: 'engine/entity/useEntityTransform',
+  category: NodeCategory.Event,
+  label: 'Use entity transform',
+  in: {
+    entity: 'entity'
+  },
+  out: {
+    positionChange: 'flow',
+    position: 'vec3',
+    rotationChange: 'flow',
+    rotation: 'quat',
+    scaleChange: 'flow',
+    scale: 'vec3'
+  },
+  initialState: initialState(),
+  init: ({ read, commit, write, graph: { getDependency } }) => {
+    const entity = Number(read('entity')) as Entity
+    const prevTransform = {}
+    const systemUUID = defineSystem({
+      uuid: 'behave-graph-useTransform-' + uniqueId(),
+      insert: { with: InputSystemGroup },
+      execute: () => {},
+      reactor: () => {
+        const transformState = useComponent(entity, TransformComponent)
+        Object.entries(transformState.value).forEach(([key, value]) => {
+          if (!Object.keys(useEntityTransform.out).includes(key)) return
+          useEffect(() => {
+            write(key as any, value)
+            commit(`${key}Change` as any)
+          }, [transformState[key]])
+        })
+        return null
+      }
+    })
+    const state: State = {
+      systemUUID
+    }
+    return state
+  },
+  dispose: ({ state: { systemUUID }, graph: { getDependency } }) => {
+    destroySystem(systemUUID)
+    return initialState()
+  }
+})
+
+export const getSelfEntity = makeFunctionNodeDefinition({
+  typeName: 'engine/entity/getSelfEntity',
+  category: NodeCategory.Query,
+  label: 'Get self entity',
+  in: {},
+  out: { entity: 'entity' },
+  exec: ({ read, write, graph: { getDependency } }) => {
+    const useEntityReactor: any = defineSystem({
+      uuid: 'behave-graph-getEntityReactor-' + uniqueId(),
+      insert: { after: PresentationSystemGroup },
+      execute: () => {},
+      reactor: function () {
+        const entity = useEntityContext()
```

Comment:
does this actually work? 

The entity context will only exist inside of a component reactor, or if a parent reactor has defined that context. 

Can we write tests for these behave-graph nodes so that we know they work correctly?